docroot/modules/custom/sitenow_events/sitenow_events.module

```diff
@@ -214,18 +231,125 @@ function sitenow_events_preprocess_block(&$variables) {
+
+        foreach ($data['events'] as $key => $value) {
           // The API nests each event as an array.
           $event = $value['event'];
           $event['heading_size'] = $heading_size ?? 'h3';
-          // Pass the parent block's layout_builder_styles to each item.
-          if (!empty($parent_lb_styles)) {
-            $event['parent_lb_styles'] = $parent_lb_styles;
+
+          $external_link = ($config->get('event_link') === 'event-link-external');
+
+          if ($external_link) {
+            $event['url'] = $event['events_site_url'];
+          }
+          else {
+            $single_event_path = $config->get('single_event_path');
+
+            // Add a parameter on the URL for event instances.
+            foreach ($event['event_instances'] as $instance_key => $instance) {
+              if ($event['instance_start'] === $instance['event_instance']['start']) {
+                $instance_id = $instance_key;
+              }
+            }
+
+            $event_path = isset($instance_id) ? "{$event['id']}/{$instance_id}" : $event['id'];
+            $event['url'] = Url::fromUri("internal:/{$single_event_path}/{$event_path}", ['alias' => TRUE]);
           }
-          $variables['content'][] = [
-            '#theme' => 'sitenow_events_teaser',
-            '#event' => $event,
+
+          $event['date_string'] = sitenow_events_format_instance([
+            'start' => $event['instance_start'],
+            'end' => $event['instance_end'],
+            'last_date' => $event['last_date'],
+            'all_day' => $event['instance_all_day'],
+          ]);
+
+          $variables['content'][$key] = [
+            '#type' => 'card',
+            '#attributes' => $attributes,
+            '#title' => $event['title'],
+            '#subtitle' => $event['date_string'],
+            '#url' => $event['url'],
+            '#headline_level' => $event['heading_size'],
           ];
+          if (isset($variables['event']['media'][0]['original_image'])) {
+            // Establish default view mode if layout_builder_styles don't exist.
+            $image_view_mode = 'large__square';
+
+            // Map layout builder styles to the view mode that should be used.
+            $media_formats = [
+              'media--widescreen' => 'large__widescreen',
+              'media--square' => 'large__square',
+              'media--circle' => 'large__square',
+            ];
+            // Loop through the map to check if any of them are being used and
+            // adjust the view mode accordingly. Last one wins out.
+            if (isset($parent_lb_styles)) {
+              foreach ($media_formats as $style => $view_mode) {
+                if (in_array($style, $parent_lb_styles)) {
+                  $image_view_mode = $view_mode;
+                }
+              }
+            }
+            // @todo Not currently working because it is expecting a URL?
+            $variables['content'][$key]['#media'] = [
```

Comment:
Wonder if there are ideas on how to pass a UI event image render array for card media. I think previously we identified different media types (image, video) going in here, but what about remote images?

Reply:
i know i have currently broke v2 ui events with [this commit](https://github.com/uiowa/uiowa/pull/6054/commits/b63a7fc66a678d66b752b92a74c16c6cfff9b9ab). i will probably call this via helper function and pass in the v2-isms.

Reply:
turns out my if statement was wrong. cleaned things up a bit.